### ISSUE TITLE
[Eclipse] Analysis errors on massively parallel execution

### ISSUE THREAD

**Issue by iloveeclipse** [spotbugs/spotbugs#3785]
Using latest Spotbugs 4.9.8 release on Java 21.0.8 JVM.

We have ~200 projects (compiled at Java 17 / 21 targets) in the Eclipse workspace and if I try to run analysis on all of them (which creates a queue of ~200 tasks where only 32 are executed in parallel), I see dataflow errors reported.

If I execute analysis on a small subset of them (10 to 20), no errors reported. So it seems that if we run lot of SpotBugs analysis tasks in parallel, there is a high chance that some not multi-threaded code will run into troubles. Last time I've tried that was on 4.9.4 snapshot (the one which had Java 25 / BCEL 6.11 snapshot) and there were no issues.

Typically suc kind of errors appear if something like static maps/fields used to cache some analysis data cause MT issues. Also possible we have problem with BCEL, to be analyzed.

Here are some of the errors I see, they are all *very* similar (just the class to analyze changes), so I assume one detector or base class or BCEL has a regression. 

```
!ENTRY com.github.spotbugs.plugin.eclipse 4 4 2025-10-21 16:53:22.621
!MESSAGE The following errors occurred during SpotBugs analysis:
!SUBENTRY 1 com.github.spotbugs.plugin.eclipse 4 0 2025-10-21 16:53:22.621
!MESSAGE error dumping dataflow analysis
!STACK 0
edu.umd.cs.findbugs.ba.DataflowAnalysisException: Invalid bytecode: could not analyze instr. invokespecial[183](3) 37 at frame []
	at edu.umd.cs.findbugs.ba.AbstractFrameModelingVisitor.analyzeInstruction(AbstractFrameModelingVisitor.java:87)
	at edu.umd.cs.findbugs.ba.type.TypeFrameModelingVisitor.analyzeInstruction(TypeFrameModelingVisitor.java:197)
	at edu.umd.cs.findbugs.ba.type.TypeAnalysis.transferInstruction(TypeAnalysis.java:407)
	at edu.umd.cs.findbugs.ba.type.TypeAnalysis.transferInstruction(TypeAnalysis.java:87)
	at edu.umd.cs.findbugs.ba.AbstractDataflowAnalysis.transfer(AbstractDataflowAnalysis.java:136)
	at edu.umd.cs.findbugs.ba.type.TypeAnalysis.transfer(TypeAnalysis.java:415)
	at edu.umd.cs.findbugs.ba.type.TypeAnalysis.transfer(TypeAnalysis.java:87)
	at edu.umd.cs.findbugs.ba.AbstractDataflowAnalysis.getFactAtLocation(AbstractDataflowAnalysis.java:84)
	at edu.umd.cs.findbugs.ba.AbstractDataflowAnalysis.getFactAfterLocation(AbstractDataflowAnalysis.java:105)
	at edu.umd.cs.findbugs.ba.Dataflow.getFactAfterLocation(Dataflow.java:517)
	at edu.umd.cs.findbugs.ba.Dataflow.dumpDataflow(Dataflow.java:571)
	at edu.umd.cs.findbugs.ba.Dataflow.execute(Dataflow.java:226)
	at edu.umd.cs.findbugs.classfile.engine.bcel.TypeDataflowFactory.analyze(TypeDataflowFactory.java:83)
	at edu.umd.cs.findbugs.classfile.engine.bcel.TypeDataflowFactory.analyze(TypeDataflowFactory.java:43)
	at edu.umd.cs.findbugs.classfile.impl.AnalysisCache.analyzeMethod(AnalysisCache.java:373)
	at edu.umd.cs.findbugs.classfile.impl.AnalysisCache.getMethodAnalysis(AnalysisCache.java:326)
	at edu.umd.cs.findbugs.classfile.engine.bcel.CFGFactory.analyze(CFGFactory.java:160)
	at edu.umd.cs.findbugs.classfile.engine.bcel.CFGFactory.analyze(CFGFactory.java:65)
	at edu.umd.cs.findbugs.classfile.impl.AnalysisCache.analyzeMethod(AnalysisCache.java:373)
	at edu.umd.cs.findbugs.classfile.impl.AnalysisCache.getMethodAnalysis(AnalysisCache.java:326)
	at edu.umd.cs.findbugs.ba.ClassContext.getMethodAnalysis(ClassContext.java:1005)
	at edu.umd.cs.findbugs.ba.ClassContext.getMethodAnalysisNoDataflowAnalysisException(ClassContext.java:990)
	at edu.umd.cs.findbugs.ba.ClassContext.getCFG(ClassContext.java:304)
	at edu.umd.cs.findbugs.detect.FindUseOfNonSerializableValue.analyzeMethod(FindUseOfNonSerializableValue.java:145)
	at edu.umd.cs.findbugs.detect.FindUseOfNonSerializableValue.visitClassContext(FindUseOfNonSerializableValue.java:96)
	at edu.umd.cs.findbugs.DetectorToDetector2Adapter.visitClass(DetectorToDetector2Adapter.java:76)
	at edu.umd.cs.findbugs.FindBugs2.lambda$analyzeApplication$1(FindBugs2.java:1111)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:317)
	at edu.umd.cs.findbugs.CurrentThreadExecutorService.execute(CurrentThreadExecutorService.java:86)
	at java.base/java.util.concurrent.AbstractExecutorService.invokeAll(AbstractExecutorService.java:247)
	at edu.umd.cs.findbugs.FindBugs2.analyzeApplication(FindBugs2.java:1121)
	at edu.umd.cs.findbugs.FindBugs2.execute(FindBugs2.java:311)
	at de.tobject.findbugs.builder.FindBugsWorker.runFindBugs(FindBugsWorker.java:316)
	at de.tobject.findbugs.builder.FindBugsWorker.work(FindBugsWorker.java:222)
	at de.tobject.findbugs.actions.FindBugsAction$StartedFromViewJob.runWithProgress(FindBugsAction.java:271)
	at de.tobject.findbugs.FindBugsJob.run(FindBugsJob.java:136)
	at org.eclipse.core.internal.jobs.Worker.run(Worker.java:63)
Caused by: edu.umd.cs.findbugs.ba.InvalidBytecodeException: Stack underflow for invokespecial[183](3) 37, 1 needed, 0 avail, frame is []
	at edu.umd.cs.findbugs.ba.type.TypeFrameModelingVisitor.consumeStack(TypeFrameModelingVisitor.java:223)
	at edu.umd.cs.findbugs.ba.type.TypeFrameModelingVisitor.visitInvokeInstructionCommon(TypeFrameModelingVisitor.java:513)
	at edu.umd.cs.findbugs.ba.type.TypeFrameModelingVisitor.visitINVOKESPECIAL(TypeFrameModelingVisitor.java:432)
	at org.apache.bcel.generic.INVOKESPECIAL.accept(INVOKESPECIAL.java:66)
	at edu.umd.cs.findbugs.ba.AbstractFrameModelingVisitor.analyzeInstruction(AbstractFrameModelingVisitor.java:84)
	... 36 more
```

```
!SUBENTRY 1 com.github.spotbugs.plugin.eclipse 4 0 2025-10-21 16:53:22.621
!MESSAGE Problem analyzing call to com.google.protobuf.GeneratedMessage$FieldAccessorTable.ensureFieldAccessorsInitialized(Ljava/lang/Class;Ljava/lang/Class;)Lcom/google/protobuf/GeneratedMessage$FieldAccessorTable;
!STACK 0
edu.umd.cs.findbugs.ba.DataflowAnalysisException: not enough values on stack: access=2, avail=0
	at edu.umd.cs.findbugs.ba.Frame.getStackValue(Frame.java:247)
	at edu.umd.cs.findbugs.ba.Hierarchy2.resolveMethodCallTargets(Hierarchy2.java:331)
	at edu.umd.cs.findbugs.ba.type.TypeFrameModelingVisitor.visitInvokeInstructionCommon(TypeFrameModelingVisitor.java:658)
	at edu.umd.cs.findbugs.ba.type.TypeFrameModelingVisitor.visitINVOKEVIRTUAL(TypeFrameModelingVisitor.java:442)
	at org.apache.bcel.generic.INVOKEVIRTUAL.accept(INVOKEVIRTUAL.java:65)
	at edu.umd.cs.findbugs.ba.AbstractFrameModelingVisitor.analyzeInstruction(AbstractFrameModelingVisitor.java:84)
	at edu.umd.cs.findbugs.ba.type.TypeFrameModelingVisitor.analyzeInstruction(TypeFrameModelingVisitor.java:197)
	at edu.umd.cs.findbugs.ba.type.TypeAnalysis.transferInstruction(TypeAnalysis.java:407)
	at edu.umd.cs.findbugs.ba.type.TypeAnalysis.transferInstruction(TypeAnalysis.java:87)
	at edu.umd.cs.findbugs.ba.AbstractDataflowAnalysis.transfer(AbstractDataflowAnalysis.java:136)
	at edu.umd.cs.findbugs.ba.type.TypeAnalysis.transfer(TypeAnalysis.java:415)
	at edu.umd.cs.findbugs.ba.type.TypeAnalysis.transfer(TypeAnalysis.java:87)
	at edu.umd.cs.findbugs.ba.AbstractDataflowAnalysis.getFactAtLocation(AbstractDataflowAnalysis.java:84)
	at edu.umd.cs.findbugs.ba.AbstractDataflowAnalysis.getFactAfterLocation(AbstractDataflowAnalysis.java:105)
	at edu.umd.cs.findbugs.ba.Dataflow.getFactAfterLocation(Dataflow.java:517)
	at edu.umd.cs.findbugs.ba.Dataflow.dumpDataflow(Dataflow.java:571)
	at edu.umd.cs.findbugs.ba.Dataflow.execute(Dataflow.java:226)
	at edu.umd.cs.findbugs.classfile.engine.bcel.TypeDataflowFactory.analyze(TypeDataflowFactory.java:83)
	at edu.umd.cs.findbugs.classfile.engine.bcel.TypeDataflowFactory.analyze(TypeDataflowFactory.java:43)
	at edu.umd.cs.findbugs.classfile.impl.AnalysisCache.analyzeMethod(AnalysisCache.java:373)
	at edu.umd.cs.findbugs.classfile.impl.AnalysisCache.getMethodAnalysis(AnalysisCache.java:326)
	at edu.umd.cs.findbugs.classfile.engine.bcel.CFGFactory.analyze(CFGFactory.java:160)
	at edu.umd.cs.findbugs.classfile.engine.bcel.CFGFactory.analyze(CFGFactory.java:65)
	at edu.umd.cs.findbugs.classfile.impl.AnalysisCache.analyzeMethod(AnalysisCache.java:373)
	at edu.umd.cs.findbugs.classfile.impl.AnalysisCache.getMethodAnalysis(AnalysisCache.java:326)
	at edu.umd.cs.findbugs.ba.ClassContext.getMethodAnalysis(ClassContext.java:1005)
	at edu.umd.cs.findbugs.ba.ClassContext.getMethodAnalysisNoDataflowAnalysisException(ClassContext.java:990)
	at edu.umd.cs.findbugs.ba.ClassContext.getCFG(ClassContext.java:304)
	at edu.umd.cs.findbugs.detect.FindUseOfNonSerializableValue.analyzeMethod(FindUseOfNonSerializableValue.java:145)
	at edu.umd.cs.findbugs.detect.FindUseOfNonSerializableValue.visitClassContext(FindUseOfNonSerializableValue.java:96)
	at edu.umd.cs.findbugs.DetectorToDetector2Adapter.visitClass(DetectorToDetector2Adapter.java:76)
	at edu.umd.cs.findbugs.FindBugs2.lambda$analyzeApplication$1(FindBugs2.java:1111)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:317)
	at edu.umd.cs.findbugs.CurrentThreadExecutorService.execute(CurrentThreadExecutorService.java:86)
	at java.base/java.util.concurrent.AbstractExecutorService.invokeAll(AbstractExecutorService.java:247)
	at edu.umd.cs.findbugs.FindBugs2.analyzeApplication(FindBugs2.java:1121)
	at edu.umd.cs.findbugs.FindBugs2.execute(FindBugs2.java:311)
	at de.tobject.findbugs.builder.FindBugsWorker.runFindBugs(FindBugsWorker.java:316)
	at de.tobject.findbugs.builder.FindBugsWorker.work(FindBugsWorker.java:222)
	at de.tobject.findbugs.actions.FindBugsAction$StartedFromViewJob.runWithProgress(FindBugsAction.java:271)
	at de.tobject.findbugs.FindBugsJob.run(FindBugsJob.java:136)
	at org.eclipse.core.internal.jobs.Worker.run(Worker.java:63)
```

The stacks are pretty similar, there are various detectors affected, here sorted list (not exclusive, there might be more, I just grepped through subset of error logs):

* CallToUnsupportedMethod
* DumbMethodInvocations
* FindBadCast2
* FindDeadLocalStores
* FindInconsistentSync2
* FindJSR166LockMonitorenter
* FindMismatchedWaitOrNotify
* FindNonSerializableValuePassedToWriteObject
* FindNullDeref
* FindOpenStream
* FindRefComparison
* FindSelfComparison2
* FindSleepWithLockHeld
* FindUnrelatedTypesInGenericContainer
* FindUnreleasedLock
* FindUnsatisfiedObligation
* FindUselessObjects
* FindUseOfNonSerializableValue
* LoadOfKnownNullValue
* NoteNonnullReturnValues
* NoteUnconditionalParamDerefs
* RedundantConditions
* SharedVariableAtomicityDetector

I will try to provide some kind of reproducer or nail down the root cause / bisect where it was introduced.

**Comment by iloveeclipse** [comment]
> Typically suc kind of errors appear if something like static maps/fields used to cache some analysis data cause MT issues. 

I see that **static** `edu.umd.cs.findbugs.ba.Dataflow.DEBUG` is modified at runtime if it exceeds 97 irterations (see  `edu.umd.cs.findbugs.ba.Dataflow.MAX_ITERS`), but I believe this is only as a result of the previous issues with code analysis.

I will try to find time to debug / bisect that in the next days. Just confirmed that 4.9.4 snapshot works fine, so it must be recent regression.
